### PR TITLE
Diagnostics update

### PR DESCRIPTION
**Pull request by civerachb-cpr**
Keep common diagnostics configurations in one file, with Dingo-O and Dingo-D specific changes (e.g. CAN IDs for the wheels) in separate files loaded dynamically.

Refactor the diagnostic updater slightly to make voltage, temperature, and amperage warning levels easier to find & edit in future.  Put a simple message in the Power Consumption topic to suppress a perceived error in robot_monitor that flags no-message as an error.  (Placeholder message just says if the robot is on battery vs shore power.)

Current issues this PR does not resolve: /imu/data_raw appears to be publishing at 2x the proper rate; Lindy's user 12V supply appears to be broken, so I'm getting a consistent "check fuse" warning on that message.  Otherwise everything looks OK in my testing.

### CODE REVIEW

File: dingo_base/src/dingo_diagnostic_updater.cpp [hunk 1]
```diff
@@ -39,6 +39,25 @@
 #include "diagnostic_updater/update_functions.h"
 #include "dingo_base/dingo_diagnostic_updater.h"
 
+// battery & user voltage warning levels
+#define VOLTAGE_ABSENT           1.0
```
Comment (tonybaltovski) [inline dingo_base/src/dingo_diagnostic_updater.cpp]: I would favour `constexpr` over macros.

**Review by tonybaltovski — CHANGES_REQUESTED** [review]

**Review by tonybaltovski — APPROVED** [review]

**Review by tonybaltovski — CHANGES_REQUESTED** [review]
Could you get the values for the TODOs before merging?

**Comment by civerachb-cpr** [comment]
Sure. Waiting on Dana to get me those numbers, then  I'll finish the merge.

**Comment by civerachb-cpr** [comment]
Requesting a re-review since there have been lots of changes related to cleaning up the lingering TODO's.
Lithium battery implementation is untested, so that's still got TODO indicators. But otherwise I think everything should be ready to be merged.

File: dingo_base/include/dingo_base/dingo_power_levels.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  // battery voltage warning levels
+  static constexpr float VOLTAGE_ABSENT                =  1.0;
+
+  // Lithium battery warning levels
+  // TODO(civerachb) -- All lithium battery-related code is currently untested, as the hardware to test it on
+  // doesn't exist yet!  We'll make any necessary changes to lithium battery support at a later date
+  static constexpr float BATTERY_LITHIUM_OVER_VOLT     = 16.9;
+  static constexpr float BATTERY_LITHIUM_LOW_VOLT      = 12.0;
+  static constexpr float BATTERY_LITHIUM_CRITICAL_VOLT = 11.2;
+  static constexpr float BATTERY_LITHIUM_LOW_PERCENT   =  0.10;    // 0-1 TODO(civerachb) -- test this out on an actual Li battery & adjust
```
Comment (tonybaltovski) [inline dingo_base/include/dingo_base/dingo_power_levels.h]: Let's change this to 0.20.

File: dingo_base/include/dingo_base/dingo_power_levels.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  static constexpr float BATTERY_LITHIUM_CRITICAL_PERCENT = 0.05;  // 0-1 TODO(civerachb) -- test this out on an actual Li battery & adjust
```
Comment (tonybaltovski) [inline dingo_base/include/dingo_base/dingo_power_levels.h]: Let's change this to 0.10

**Review by tonybaltovski — CHANGES_REQUESTED** [review]
Two minor nits but looks good.  The wireless connected was removed in 3505dd25cf613c0960ad82936746eaa5aa62376f, any reason for that?

File: dingo_base/include/dingo_base/dingo_power_levels.h (hunk 2 above)
Comment (civerachb-cpr) [reply dingo_base/include/dingo_base/dingo_power_levels.h]: I'd added the wi-fi connected entry in an earlier commit, but it was causing an error when I tested it, so I removed it again.
I'll increase the low threshold to 0.2.

**Review by tonybaltovski — COMMENTED** [review]
Thanks!

**Pull request merged**